src/core/portal/portal-directives.ts

```diff
@@ -0,0 +1,110 @@
+import {Portal} from './portal';
+import {TemplatePortal} from './portal';
+import {ComponentRef} from 'angular2/core';
+import {ComponentPortal} from './portal';
+import {Directive} from 'angular2/core';
+import {TemplateRef} from 'angular2/core';
+import {BasePortalHost} from './portal';
+import {DynamicComponentLoader} from 'angular2/core';
+import {ElementRef} from 'angular2/core';
+import {ViewContainerRef} from 'angular2/core';
+import {PromiseWrapper} from 'angular2/src/facade/promise';
+
+
+/**
+ * Directive version of a `TemplatePortal`. Because the directive *is* a TemplatePortal,
+ * the directive instance itself can be attached to a host, enabling declarative use of portals.
+ *
+ * Usage:
+ * <template portal #greeting>
+ *   <p> Hello {{name}} </p>
+ * </template>
+ */
+@Directive({
+  selector: '[portal]',
+  exportAs: 'portal',
+})
+export class TemplatePortalDirective extends TemplatePortal {
+  constructor(templateRef: TemplateRef) {
+    super(templateRef);
+  }
+}
+
+
+/**
+ * Directive version of a PortalHost. Because the directive *is* a PortalHost, portals can be
+ * directly attached to it, enabling declarative use.
+ *
+ * Usage:
+ * <template [portalHost]="greeting"></template>
+ */
+@Directive({
+  selector: '[portalHost]',
+  inputs: ['portal: portalHost']
+})
+export class PortalHostDirective extends BasePortalHost {
+  /** The attached portal. */
+  private portal_: Portal<any>;
+
+  constructor(
+      private dynamicComponentLoader: DynamicComponentLoader,
+      private elementRef_: ElementRef,
+      private viewContainerRef_: ViewContainerRef) {
+    super();
+  }
+
+  get portal(): Portal<any> {
+    return this.portal_;
+  }
+
+  set portal(p: Portal<any>) {
+    this.replaceAttachedPortal_(p);
+  }
+
+  /** Attach the given ComponentPortal to this PortlHost using the DynamicComponentLoader. */
+  attachComponentPortal(portal: ComponentPortal): Promise<ComponentRef> {
+    portal.setAttachedHost(this);
+
+    // If the portal specifies an origin, use that as the logical location of the component
+    // in the application tree. Otherwise use the location of this PortalHost.
+    let elementRef = portal.origin != null ? portal.origin : this.elementRef_;
+
+    // Typecast is necessary for Dart transpilation.
+    return <Promise<ComponentRef>>
```

Comment:
It seems like this cast is unnecessary, as `loadNextToLocation` already returns this.